x-pack/plugins/maps/public/classes/sources/esql_source/esql_utils.ts

```diff
@@ -88,31 +87,18 @@ export function getFieldType(column: ESQLColumn) {
 }
 
 async function getColumns(esql: string) {
-  const params = {
-    query: esql + ' | limit 0',
-    version: ESQL_LATEST_VERSION,
-  };
-
-  try {
-    const resp = await lastValueFrom(
-      getData().search.search(
-        { params },
-        {
-          strategy: 'esql',
-        }
-      )
-    );
-
-    const searchResponse = resp.rawResponse as unknown as ESQLSearchReponse;
-    return searchResponse.all_columns ? searchResponse.all_columns : searchResponse.columns;
-  } catch (error) {
-    throw new Error(
-      i18n.translate('xpack.maps.source.esql.getColumnsErrorMsg', {
-        defaultMessage: 'Unable to load columns. {errorMessage}',
-        values: { errorMessage: error.message },
-      })
-    );
-  }
+  const columns = await getESQLQueryColumns({
+    esqlQuery: esql,
+    search: getData().search.search,
+  });
+  return columns.map((col) => {
```

Comment:
how many columns can be returned from ESQL? Should we be concerned about performance if 10000K columns are returned? Would it be possible to have `getESQLQueryColumnsRaw` and `getESQLQueryColumns`? `getESQLQueryColumnsRaw` would return just return `rawResponse.columns` and `getESQLQueryColumns` would just return formatted response from `getESQLQueryColumns`

Seems inefficient to convert response to one format, just to convert it back to the original format.

Reply:
It won't be, we are not expecting more than 10K columns. But in my point of view the more the utils we have, the merrier. I applied your suggestion here https://github.com/elastic/kibana/pull/182691/commits/4e3197da95172364cba5811118c77344b8d14b44